### PR TITLE
Project advisers

### PR DESCRIPTION
**Pull request by g4s8**
#546 - advisers in catalog + `@todo`

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#827](https://github.com/zerocracy/farm/issues/827) is now in scope, role is `REV`

File: src/main/java/com/zerocracy/pmo/Catalog.java [hunk 1]
```diff
@@ -136,7 +136,8 @@ public void add(final String pid, final String prefix) throws IOException {
                     .add("prefix").set(prefix).up()
                     .add("alive").set(true).up()
                     .add("fee").set(Cash.ZERO).up()
-                    .add("publish").set(Boolean.toString(false))
+                    .add("publish").set(Boolean.toString(false)).up()
+                    .add("adviser").set("0crat").up()
```
Comment (carlosmiranda) [inline src/main/java/com/zerocracy/pmo/Catalog.java]: @g4s8 minor quibble... do we need call to `up()` here? We didn't have it as a final directive before.

File: src/main/java/com/zerocracy/pmo/Catalog.java [hunk 2]
```diff
@@ -567,6 +568,57 @@ public String title(final String pid) throws IOException {
...
+     * Project adviser.
+     * @param pid Project id
+     * @return Adviser id
+     * @throws IOException If fails
+     */
+    public String adviser(final String pid) throws IOException {
+        if (!this.exists(pid)) {
+            throw new IllegalArgumentException(
+                new Par(
+                    "Project %s doesn't exist, can't get title"
```
Comment (carlosmiranda) [inline src/main/java/com/zerocracy/pmo/Catalog.java]: @g4s8 Perhaps you mean "can't get adviser"? :)

File: src/main/java/com/zerocracy/pmo/Catalog.java [hunk 3]
```diff
@@ -567,6 +568,57 @@ public String title(final String pid) throws IOException {
...
+     * @param pid Project id
+     * @param adviser Adviser id
+     * @throws IOException If fails
+     */
+    public void adviser(final String pid, final String adviser)
+        throws IOException {
+        if (!this.exists(pid)) {
+            throw new IllegalArgumentException(
+                new Par(
+                    "Project %s doesn't exist, can't get title"
```
Comment (carlosmiranda) [inline src/main/java/com/zerocracy/pmo/Catalog.java]: @g4s8 same here, should say "can't get adviser".

**Comment by g4s8** [comment]
@carlosmiranda fixed, please look again

**Comment by carlosmiranda** [comment]
@yegor256 looks good to merge

**Comment by yegor256** [comment]
@rultor merge

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#827](https://github.com/zerocracy/farm/issues/827) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 points just awarded to @yegor256[/z](https://www.0crat.com/u/yegor256)

**Comment by ypshenychka** [comment]
@carlosmiranda According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>The code reviewer found at least three problems in the code.
Comments were mostly about design problems, not cosmetic issues.

Two issues were found during code review. 
Please confirm that you'll try to find at least three problems while future reviews.

**Comment by carlosmiranda** [comment]
@ypshenychka confirmed

**Comment by ypshenychka** [comment]
@carlosmiranda thanks

**Comment by ypshenychka** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
Order was finished, quality is "acceptable": +15 points just awarded to @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda)

**Comment by 0crat** [comment]
Quality review completed: +8 points just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Pull request merged**